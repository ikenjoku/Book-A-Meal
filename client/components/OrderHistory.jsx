import React, { Component } from 'react';

import NavBar from './NavBar.jsx';

import { sampleOrders } from "./mocks";

export default class OrderHistory extends Component{
  constructor(props){
    super(props);
    this.state = {
      orders: sampleOrders.orders
    }
  }


  render(){
    return (
      <div className="contain-page">
      <NavBar/>
      <main id='order-history-container' className="order-history-content">
          <div className="cool-lg-text">
              <h2>Order History</h2>
          </div>
          <OrderSearch/>
          <OrderList orders={this.state.orders}/>
      </main>
      <footer className="footer order-history-footer">
          <p>Book-A-Meal &copy;2018</p>
      </footer>
  </div>
    );
  }
}


class OrderSearch extends Component{
  constructor(props){
    super(props);
  }

  render(){
    return (
      <div className="order-history-date-picker center">
        <form>
          <input 
            className="input-field-control" 
            type="date" 
            placeholder="Choose a date..."  
            name="history-date" />
          <input 
            className="input-btn-control" 
            type="submit" 
            value="Choose date" />
        </form>
    </div>
    );
  }
}

const OrderList = ({orders}) => {

  return (
    <div>
      <div className="order-table-header">
        <p>Order ID</p>
        <p>Customer</p>
        <p>Order Summary</p>
        <p>Amount</p>
      </div>
      <div className="order-table-body">
        { orders.map(order =>
          <div key={order.id} className="order-table-body-row">
              <p>{order.id}</p>
              <p>{order.User.firstname} {order.User.lastname}</p>
              <div className="order-tbody-summary">
                <p>{order.Meal.name}</p>
              </div>
              <p>&#8358;{order.amount}</p>
          </div>
        ) }
      </div>
      <div id="total-made">
          <h2>Total: &#8358;150 470.00</h2>
      </div>
  </div>
  );
}