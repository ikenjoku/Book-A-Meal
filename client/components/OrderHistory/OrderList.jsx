import React, { Component } from 'react';
import { connect } from 'react-redux';
import Loader from 'react-loader-spinner';
import { deliverAnOrder } from "../../actions/orderActions";

const OrderList = ({ orders, deliverAnOrder, isLoadingOrders }) => {
  let daysTotal = 0;
  let serialNum = 0
  orders.map(order =>{ if(order.status === 'delivered')  daysTotal += order.amount});
  const renderedJsx = orders.length === 0 ? 
  <p className='error-alert'>No Orders found for this day</p> :
  <div>
  <table className="order-table-body">
        <thead>
          <tr>
            <th>S/N</th>
            <th>Customer</th>
            <th>Order Summary</th>
            <th>Amount</th>
            <th>Status</th>
            <th>Action</th>
          </tr>
        </thead>
          <tbody>
          { orders.map(order =>
            <tr key={order.id}>
              <td >{++serialNum}</td>
              <td>{order.User.firstname} {order.User.lastname}</td>
              <td>{order.Meal.name}</td>
              <td>&#8358;{order.amount}</td>
              <td>{order.status}</td>
              <td>{order.status === 'pending'  && <button className='deliverBtn' onClick={() => deliverAnOrder(order.id)}>deliver</button>}</td>
            </tr>
            )}
          </tbody>          
      </table>
      <div id="total-made">
        <h2>Total: &#8358;{daysTotal}</h2>
      </div>
      </div>
  
  return (
    <div>
      {
        isLoadingOrders ? 
        <div className='loading-spinner'>
          <Loader 
              type="Circles"
              color="#9D2401"
              height="100"
              width="100"
          />
          <h3 id='loader-text'>Fetching...</h3>  
          </div> : renderedJsx
      }
    </div>
  );
}

const mapStateToProps = state => {
  return{
    orders: state.orderReducer.orders,
    isLoadingOrders: state.orderReducer.isLoadingOrders,
    error: state.orderReducer.error
  }
}
export default connect(mapStateToProps, { deliverAnOrder })(OrderList);