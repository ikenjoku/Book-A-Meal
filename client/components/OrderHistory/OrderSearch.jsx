import React, { Component } from 'react';


export default class OrderSearch extends Component {
  constructor(props) {
    super(props);
  }

  render() {
    return (
      <div className="order-history-date-picker center">
        <form>
          <input
            className="input-field-control"
            type="date"
            placeholder="Choose a date..."
            name="history-date" />
          <input
            className="input-btn-control"
            type="submit"
            value="Choose date" />
        </form>
      </div>
    );
  }
}
