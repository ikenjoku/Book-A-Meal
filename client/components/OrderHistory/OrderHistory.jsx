import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { connect } from "react-redux";
import DatePicker from '../DatePicker.jsx';
import OrderList from './OrderList.jsx';
import { getOrdersByDate } from "../../actions/orderActions";


class OrderHistory extends Component {
  constructor(props) {
    super(props);

  }
  componentDidMount() {
    const today = new Date().toISOString().substr(0, 10);
    this.props.getOrdersByDate({ selectedDate: today });
  }

  render() {
    return (
      <main id='order-history-container' className="order-history-content">
        <div className="cool-lg-text">
          <h2>Order History</h2>
        </div>
        <DatePicker btnName='Get Orders' onSubmit={
          (selectedDate) => this.props.getOrdersByDate(selectedDate)
        } />
        <OrderList />
      </main>
    );
  }
}

export default connect(null, { getOrdersByDate })(OrderHistory);
