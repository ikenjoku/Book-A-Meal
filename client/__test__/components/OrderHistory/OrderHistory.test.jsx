import React from 'react';
import { shallow } from 'enzyme';
import { OrderHistory } from '../../../components/OrderHistory/OrderHistory';

describe('OrderHistory Component', () => {
  let wrapper;
  const getOrdersByDate = jest.fn();

  test('should render the OrderHistory correctly', () => {
    wrapper = shallow(<OrderHistory getOrdersByDate={getOrdersByDate} />);
    expect(wrapper).toMatchSnapshot();
  });

  test('DatePicker should handle onSubmit', () => {
    wrapper.find('DatePicker').prop('onSubmit')('2018-08-08');
    expect(getOrdersByDate).toHaveBeenLastCalledWith('2018-08-08');
  });
});
