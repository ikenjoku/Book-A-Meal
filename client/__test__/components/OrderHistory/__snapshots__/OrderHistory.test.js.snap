// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`OrderHistory Component should render the OrderHistory correctly 1`] = `
<main
  className="order-history-content"
  id="order-history-container"
>
  <div
    className="cool-lg-text"
  >
    <h2>
      Order History
    </h2>
  </div>
  <DatePicker
    btnName="Get Orders"
    onSubmit={[Function]}
  />
  <Connect(OrderList) />
</main>
`;
